File: php/utility/PrepareBody.php
<?php
declare(strict_types=1);

// ManualPrecipitationStations SDK utility: prepare_body

class ManualPrecipitationStationsPrepareBody
{
    public static function call(ManualPrecipitationStationsContext $ctx): mixed
    {
        if ($ctx->op->input === 'data') {
            return ($ctx->utility->transform_request)($ctx);
        }
        return null;
    }
}

File: go/utility/prepare_body.go
package utility

import "github.com/voxgig-sdk/manual-precipitation-stations-sdk/go/core"

func prepareBodyUtil(ctx *core.Context) any {
	op := ctx.Op

	if op.Input == "data" {
		body := ctx.Utility.TransformRequest(ctx)
		return body
	}

	return nil
}
